File: test/src/main/java/com/consultant/domain/sys/Role.java
package com.consultant.domain.sys;

import org.spin.data.core.AbstractEntity;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

/**
 * 角色
 * <p>Created by xuweinan on 2017/04/20.</p>
 *
 * @author xuweinan
 */
@Entity
@Table(name = "sys_role")
public class Role extends AbstractEntity {

    @Column(length = 64)
    private String name;

    @Column(length = 64)
    private String code;

    @Column
    private String remark;

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(name = "sys_role_permission", joinColumns = @JoinColumn(name = "role_id"), inverseJoinColumns = @JoinColumn(name = "permission_id"))
    private List<Permission> permissions = new ArrayList<>();

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public String getRemark() {
        return remark;
    }

    public void setRemark(String remark) {
        this.remark = remark;
    }

    public List<Permission> getPermissions() {
        return permissions;
    }

    public void setPermissions(List<Permission> permissions) {
        this.permissions = permissions;
    }
}

File: test/src/main/kotlin/com/shipping/domain/sys/Role.kt
package com.shipping.domain.sys

import org.spin.data.core.AbstractEntity
import java.util.*
import javax.persistence.*

/**
 * 角色
 *
 * Created by xuweinan on 2017/04/20.
 *
 * @author xuweinan
 */
@Entity
@Table(name = "sys_role")
class Role(
        @Column(length = 64, unique = true)
        var name: String = "",

        @Column(length = 64, unique = true)
        var code: String = "",

        @ManyToMany(fetch = FetchType.LAZY)
        @JoinTable(name = "sys_role_permission", joinColumns = [JoinColumn(name = "role_id")], inverseJoinColumns = [JoinColumn(name = "permission_id")])
        var permissions: MutableList<Permission> = ArrayList()
) : AbstractEntity() {
    companion object {
        private const val serialVersionUID = 8934787783435264166L
    }
}
